src/lib/repl.js

```diff
@@ -0,0 +1,30 @@
+import { options } from 'preact';
+import { lazy } from 'preact-iso';
+
+const CodeEditor = lazy(() => import('../components/code-editor'));
+const Runner = lazy(() => import('../components/controllers/repl/runner'));
+
+// `preact-iso` doesn't forward refs to lazy-loaded components (should we?)
+// so we need to manually do it here to support the tutorial which reads the runner ref
+const oldDiff = options.__b;
+options.__b = (vnode) => {
+	if (vnode.type === Runner && vnode.ref) {
+		vnode.props.ref = vnode.ref;
+		vnode.ref = null;
+	}
+
+	if (oldDiff) oldDiff(vnode);
+};
```

Comment:
Might be worth adding this to `preact-iso`? I didn't realize we weren't forwarding refs for `lazy` there already.